lib/teiserver/o_auth/queries/application_query.ex

```diff
@@ -88,4 +89,65 @@ defmodule Teiserver.OAuth.ApplicationQueries do
     from [app: app] in query,
       where: app.uid == ^uid
   end
+
+  @doc """
+  Returns all OAuth applications that the user has authorized (has tokens for, including expired ones).
+  """
+  @spec list_authorized_applications(T.userid()) :: [Application.t()]
+  def list_authorized_applications(user_id) do
+    base_query()
+    |> join(:left, [app], token in Teiserver.OAuth.Token,
+      on: token.application_id == app.id and token.owner_id == ^user_id
+    )
+    |> where([app, token], not is_nil(token.id))
+    |> distinct([app, token], true)
+    |> preload(:owner)
+    |> Repo.all()
+  end
+
+  @doc """
+  Returns the count of active tokens for each application that the user has authorized.
+  Counts only the user's tokens, not all tokens for the application.
+  """
+  @spec get_application_token_counts(T.userid()) :: %{
+          Application.id() => non_neg_integer()
+        }
+  def get_application_token_counts(user_id) do
+    from(token in Teiserver.OAuth.Token,
+      where: token.owner_id == ^user_id,
+      where: token.expires_at > ^DateTime.utc_now(),
+      group_by: token.application_id,
+      select: {token.application_id, count(token.id)}
+    )
+    |> Repo.all()
+    |> Enum.into(%{})
+  end
+
+  @doc """
+  Deletes all tokens (access and refresh) for a specific application and user.
+  Returns the count of deleted tokens.
+  """
+  @spec delete_user_application_tokens(T.userid(), Application.id()) ::
+          {non_neg_integer(), nil}
```

Comment:
the return type is a bit wonky no? Why returns a tuple with a nil as the second argument? It should just return the deleted count